### PR TITLE
Custom rtmp port

### PR DESCRIPTION
**Pull request by wallopthecat**
Please review this approach - still have to apply it to the rpi Dockerfiles (and someone needs to look at docker-compose as well) if you approve. This lets me use the same host for two separate streams

### CODE REVIEW

**Comment by ioppermann** [comment]
Thanks. I will check it out the next days.

File: run.sh [hunk 1]
```diff
@@ -2,6 +2,11 @@
 MODE=${MODE:="default"}
 RS_MODE=${RS_MODE:=$MODE}
 
+envsubst '$${RS_RTMP_PORT}' < /restreamer/conf/nginx.conf.template > /restreamer/conf/nginx.conf
+envsubst '$${RS_RTMP_PORT}' < /restreamer/conf/nginx_ssl.conf.template > /restreamer/conf/nginx_ssl.conf
+envsubst '$${RS_RTMP_PORT}' < /restreamer/conf/live.json.template > /restreamer/conf/live.json
+envsubst '$${RS_RTMP_PORT}' < /restreamer/conf/live-rpi.json.template > /restreamer/conf/live-rpi.json
```
Comment (wallopthecat) [inline run.sh:8]: this is where the magic happens - the rest of the changes are basically replacing 1935 with the env variable and setting a default to 1935

File: Dockerfile [hunk 2]
```diff
@@ -118,7 +120,8 @@ RUN apt-get update && \
         zlib1g \
         v4l-utils \
         libv4l-0 \
-        alsa-utils
+        alsa-utils \
+        gettext-base
```
Comment (wallopthecat) [inline Dockerfile:124]: required for `envsubst` env variable token replacement

File: Dockerfile [hunk 3]
```diff
@@ -107,6 +107,8 @@ COPY --from=builder /usr/local/bin /usr/local/bin
 COPY --from=builder /usr/local/nginx /usr/local/nginx
 COPY --from=builder /usr/local/lib /usr/local/lib
 
+ENV RS_RTMP_PORT=1935
```
Comment (wallopthecat) [inline Dockerfile:110]: sets default port as all configs now are templates with configurable RTMP port

File: run.sh (hunk 1 above)
Comment (ioppermann) [reply run.sh:9]: Can't this done with `sed` instead of pulling in the gettext-base package?

File: Dockerfile (hunk 2 above)
Comment (ioppermann) [reply Dockerfile:122]: Maybe there's a solution with `sed` instead of pulling in the gettext-base package.

**Review by ioppermann — CHANGES_REQUESTED** [review]
Thanks for the suggested changes. I added some comments about the possibilities replacing the `envsubst` tool with `sed`. This would avoid installing the `gettext-base` package.

**Comment by wallopthecat** [comment]
I'm open to using sed instead, but not sure when I will have time to do so. Thanks for your review!

**Comment by ioppermann** [comment]
Don't worry. I will also try to figure something out. But I guess it will be tricky with the `$`, `{`, and `}` ...

**Comment by ioppermann** [comment]
Can you explain me the exact use-case for a custom RTMP port? If you are using Docker to run different Restreamer instances on the same box, you can simply map the RTMP port to a different port (e.g. -p 1936:1935), the same way you do it with the HTTP port. This means the RTMP port inside the Docker container can always be the same.

**Comment by jstabenow** [comment]
Thanks for the request! We have solved it in the new version:
https://docs.datarhei.com/restreamer/knowledge-base/manual/system-settings/rtmp

**Pull request closed without merging**